code/go/pkg/validator/api_test.go

```
@@ -0,0 +1,517 @@
+	default:
+		assert.ElementsMatch(t,
+			extractMessages(legacy),
+			extractMessages(newAPI),
+			"error sets differ between legacy and new API")
+	}
+}
+
+// extractMessages unpacks a specerrors.ValidationErrors into individual
+// message strings, or wraps a plain error in a single-element slice.
+func extractMessages(err error) []string {
+	if verrs, ok := err.(specerrors.ValidationErrors); ok {
```

golangci-lint (2.12.2)

[error] type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors

(errorlint)